evalml/objectives/standard_metrics.py

```diff
@@ -827,6 +1040,48 @@ def objective_function(self, y_true, y_predicted, X=None, sample_weight=None):
         )
 
 
+class MASE(TimeSeriesRegressionObjective):
+    """Mean absolute scaled error for time series regression. Scaled by 100 to return a percentage.
+
+    Only valid if there exists a nonzero input in y_train. Otherwise, will throw a ValueError.
+
+    Example:
+        >>> y_train = np.array([5, 0.5, 4, 6, 3, 5, 2])
+        >>> y_true = np.array([3, -0.5, 2, 7, 2])
+        >>> y_pred = np.array([2.5, 0.0, 2, 8, 1.25])
+        >>> np.testing.assert_almost_equal(MASE().objective_function(y_true, y_pred, y_train), 0.18333333333333335)
+    """
+
+    name = "Mean Absolute Scaled Error"
+    greater_is_better = False
+    score_needs_proba = False
+    perfect_score = 0.0
+    is_bounded_like_percentage = False  # Range [0, Inf)
+    expected_range = [0, float("inf")]
+
+    def objective_function(
+        self,
+        y_true,
+        y_predicted,
+        y_train,
+        X=None,
+        sample_weight=None,
+    ):
+        """Objective function for mean absolute scaled error for time series regression."""
+        # [0, 1, 1, 0] -- good
+        # [0, 0, 0, 0] -- bad
+
+        if (isinstance(y_train, pd.DataFrame) and (y_train.values == 0).all()) or (
+            isinstance(y_train, pd.Series) and (y_train == 0).all()
+        ):
```

Comment:
I don't think you need to split it up this way - `y_train.values` should work regardless of whether it's a dataframe or a series!